File: leetcode_py/Path Sum.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inSum(self,root,sum):
        if root==None:
            return False
        if sum==root.val and root.left==None and root.right==None:
            return True
        left=self.inSum(root.left,sum-root.val)
        right=self.inSum(root.right,sum-root.val)
        return left|right
    # @param root, a tree node
    # @param sum, an integer
    # @return a boolean
    def hasPathSum(self, root, sum):
        return self.inSum(root,sum)

File: leetcode_js/Path Sum.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} sum
 * @return {boolean}
 */
var helper = function(root, sum){
    if(!root.left && !root.right){
        return root.val === sum;
    }else return (!!root.left && helper(root.left, sum - root.val)) ||
     (!!root.right && helper(root.right, sum - root.val));
}
var hasPathSum = function(root, sum) {
    if(!root) return false;
    return helper(root, sum);
};